### ISSUE TITLE
Depth Output 0 or Nan

### ISSUE THREAD

**Issue by KaiMJ** [ashawkey/torch-ngp#198]
I am trying to implement a depth loss, in nerf/utils.py/Trainer.train_step.

        outputs = self.model.render(rays_o, rays_d, staged=False, bg_color=bg_color, perturb=True, force_all_rays=False if self.opt.patch_size == 1 else True, **vars(self.opt))    

the outputs['depth'] always return 0 or nan during training. 

depth = outputs['depth']
print(depth[~depth.isnan()].sum()) --> always 0.

With self.model.eval(), I get an appropriate depth result with some non zero values, but unable to backprop.

Any idea what is going on?

**Comment by jaydeepradeJD** [comment]
I was facing similar issue, try increasing the bound! If ray does not hit the aabb, it assigns the very large values for near and far, that causes NaN in downstream process. Check here: https://github.com/ashawkey/torch-ngp/blob/b6e080468925f0bb44827b4f8f0ed08291dcf8a9/raymarching/src/raymarching.cu#L122
